src/server/campaignd/server.cpp

```diff
@@ -247,23 +247,19 @@ void server::load_config()
 			const std::string& addon_id = campaign["name"].str();
 			const std::string& addon_file = campaign["filename"].str();
 			if(get_addon(addon_id)) {
-				ERR_CS << "The addon '" << addon_id
-					   << "' already exists in the new form! The old version will be ignored and its file deleted!\n";
-				filesystem::delete_file(filesystem::normalize_path(addon_file));
-				continue;
+				throw filesystem::io_exception("The addon '" + addon_id
+					   + "' already exists in the new form! Possible code or filesystem interference!\n");
 			}
 			if(std::find(legacy_addons.begin(), legacy_addons.end(), addon_id) == legacy_addons.end()) {
-				ERR_CS << "No file has been found for the legacy addon '" << addon_id
-					   << "'. Please, check the file structure!\n";
-				continue;
+				throw filesystem::io_exception("No file has been found for the legacy addon '" + addon_id
+					   + "'. Please, check the file structure!\n");
```

Comment:
This doesn't matter for the PR, but in the future avoid using un-assertive language like "please" when addressing either the player or (in this case) server administrators over stuff that demands their immediate action.